File: ios/IsIosAppOnMac.m
//  IsIosAppOnMac.m
//  IsIosAppOnMac

#import <React/RCTBridgeModule.h>
#import <React/RCTLog.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(IsIosAppOnMac, NSObject)

RCT_EXTERN_METHOD(
    isIosAppOnMac:(RCTPromiseResolveBlock)success
    reject:(RCTPromiseRejectBlock)reject
);

@end

File: ios/IsIosAppOnMac.swift
import Foundation

@objc(IsIosAppOnMac)
class IsIosAppOnMac: NSObject {    
    @objc
    static func requiresMainQueueSetup() -> Bool {
        return false
    }

    @objc
    func isIosAppOnMac(
        _ success: RCTPromiseResolveBlock,
        reject: RCTPromiseRejectBlock
    ) {
        var isiOSAppOnMac = false
        if #available(iOS 14.0, *) {
            isiOSAppOnMac = ProcessInfo.processInfo.isiOSAppOnMac
        }
        success(isiOSAppOnMac)
    }
}
